test/infrastructure/inmemory/internal/controllers/inmemorymachine_controller.go

```diff
@@ -753,6 +756,117 @@ func (r *InMemoryMachineReconciler) reconcileNormalKubeadmObjects(ctx context.Co
 	return ctrl.Result{}, nil
 }
 
+func (r *InMemoryMachineReconciler) reconcileNormalKubeProxy(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, _ *infrav1.InMemoryMachine) (ctrl.Result, error) {
+	// No-op if the machine is not a control plane machine.
+	if !util.IsControlPlaneMachine(machine) {
+		return ctrl.Result{}, nil
+	}
+
+	// TODO: Add provisioning time for KubeProxy.
+
+	// Compute the resource group unique name.
+	// NOTE: We are using reconcilerGroup also as a name for the listener for sake of simplicity.
+	resourceGroup := klog.KObj(cluster).String()
+	cloudClient := r.CloudManager.GetResourceGroup(resourceGroup).GetClient()
+
+	// Create the kube-proxy-daemonset
+	kubeProxyDaemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: metav1.NamespaceSystem,
+			Name:      "kube-proxy",
+			Labels: map[string]string{
+				"component": "kube-proxy",
+			},
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "kube-proxy",
+							Image: "k8s.io/kube-proxy:v1.27.1",
```

Comment:
Would it make sense to use the version from the Machine?

I think this would be ~ how kubeadm generates it. Otherwise I think we get an additional update of this DS during cluster creation